### PR TITLE
[zxing] Initial implementation of ZXing. Contributes to JB#49377

### PR DESCRIPTION
**Pull request by HandsomeJaack**

### CODE REVIEW

File: rpm/zxing-cpp.spec [hunk 1]
```diff
@@ -0,0 +1,44 @@
+Name:       zxing-cpp
+Summary:    ZXing port to C++
+Version:    1.1.1
+Release:    1
+License:    Apache-2.0
```
Comment (pvuorela) [inline rpm/zxing-cpp.spec]: Use ASL 2.0 for the name.

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[submodule "zxing-cpp"]
+	path = zxing-cpp
+	url = https://github.com/nu-book/zxing-cpp
```
Comment (pvuorela) [inline .gitmodules]: Use
https://git.sailfishos.org/mirror/zxing-cpp

Comment (okodron) [reply .gitmodules]: I guess it's only used for packages hosted on git.sailfishos.org

Comment (pvuorela) [reply .gitmodules]: Haven't had too many mirroring projects yet here so confirmed: should use the sailfish mirror also on github repositories.

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,3 @@
+!RPMS/*
+!.vscode/*
+!.mb2/*
```
Comment (dseight) [inline .gitignore]: Why these patterns are negated?

**Comment by pvuorela** [comment]
LGTM

**Pull request merged**